pkg/egressgateway/manager.go

```diff
@@ -24,25 +25,38 @@ import (
 	"github.com/cilium/cilium/pkg/maps/egressmap"
 	nodeTypes "github.com/cilium/cilium/pkg/node/types"
 	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+type Manager interface {
+	OnAddEgressPolicy(config PolicyConfig)
```

Comment:
Are there interface definitions for these we can just embed here? e.g. `subscriber.Node`? Would make the purpose clear.